File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/data/DataInt64.java
package tasks.data;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Int64;

@JavaTaskInfo
public class DataInt64 extends JavaTask {

	@InputPortInfo(name = "In", dataType = Int64.class)
	public InputPort in;
	@OutputPortInfo(name = "Out", dataType = Int64.class)
	public OutputPort out;

	@Override
	public void execute(int grp) {
		Int64 inInt64 = in.getData(Int64.class);
		out.putData(new Int64(inInt64.getValue()*2));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/data/DataInt64.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Int64 import Int64

@PythonTaskInfo
class DataInt64(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Int64)
        self.in2 = InputPort(name="in2", data_type=Int64)
        self.out = OutputPort(name="out", data_type=Int64)

    def execute(self, grp):
        in1Data = self.in1.get_data(Int64)
        in2Data = self.in2.get_data(Int64)
        self.out.put_data(Int64(in1Data.values[0] + in2Data.values[0]))